cruise-control-core/src/main/java/com/linkedin/cruisecontrol/common/utils/AutoCloseableLock.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.common.utils;
+
+import java.util.concurrent.locks.Lock;
+
+
+public class AutoCloseableLock implements AutoCloseable {
+
+  private final Lock _lock;
+
+  public AutoCloseableLock(Lock lock) {
+    _lock = Utils.validateNotNull(lock, "Lock cannot be null");
+    _lock.lock();
+  }
+
+  @Override
+  public void close() {
+    try {
+      _lock.unlock();
+    } catch (Exception e) {
+      throw new IllegalStateException("while invoking close action", e);
+    }
+  }
```

Comment:
According to the documentation of this method:
```
...
     * However, implementers of this interface are strongly encouraged
     * to make their {@code close} methods idempotent.
...
```

But the current implementation can throw an exception if the lock has already been unlocked. Can we make this idempotent by adding ` private final AtomicBoolean _closed = new AtomicBoolean(false);` to this class and updating the close method as follows:
```
    if (_closed.compareAndSet(false, true)) {
      _lock.unlock();
    }
```

Reply:
Good idea. Fixed